src/Microsoft.FeatureManagement.Telemetry.ApplicationInsights.AspNetCore/TargetingTelemetryInitializer.cs

```diff
@@ -42,20 +40,13 @@ protected override void OnInitializeTelemetry(HttpContext httpContext, RequestTe
                 throw new ArgumentNullException(nameof(httpContext));
             }
 
-            // Extract the targeting id from the http context
-            string targetingId = null;
-
-            if (httpContext.Items.TryGetValue(TargetingIdKey, out object value))
-            {
-                targetingId = value?.ToString();
-            }
-
-            if (!string.IsNullOrEmpty(targetingId))
+            // Extract the targeting id from the request telemetry
+            if (requestTelemetry.Properties.ContainsKey(Constants.TargetingIdKey))
             {
                 // Telemetry.Properties is deprecated in favor of ISupportProperties
                 if (telemetry is ISupportProperties telemetryWithSupportProperties)
                 {
-                    telemetryWithSupportProperties.Properties["TargetingId"] = targetingId;
+                    telemetryWithSupportProperties.Properties[Constants.TargetingIdKey] = requestTelemetry.Properties[Constants.TargetingIdKey];
```

Comment:
I think we can use `Add` method here.

This is a code snippet from `AspNetCoreEnvironmentTelemetryInitializer` from `Microsoft.ApplicationInsights.AspNetCore.TelemetryInitializers`
``` C#
public void Initialize(ITelemetry telemetry)
{
    if (environment != null && telemetry is ISupportProperties supportProperties && !supportProperties.Properties.ContainsKey("AspNetCoreEnvironment"))
    {
        supportProperties.Properties.Add("AspNetCoreEnvironment", environment.EnvironmentName);
    }
}
```
Let's follow it.

Reply:
I think we have the correct usage right now. `.Add` will throw an exception if the key already exists. For our middleware, we really don't expect the key to already exist / it would be weird if it did. For this initializer, it runs on **all** telemetry, meaning it even runs on the RequestTelemetry itself. This means the key will already exist during that call.

Reply:
We shouldn't override it if someone already set it.

Reply:
@zhenlan  If someone already set it, what do you prefer happens in these cases: 
1. Developer is setting TargetingId correctly (I think throw, skip, or override are all valid)
2. Developer is setting TargetingId incorrectly. ie: to a different value than TargetingContext.UserId (I think throw or override).
3. Developer is not setting TargetingId (We add/override)

To me, override is the safest option. If they don't like it they can stop using the initializer.